cmd/cloudflared/tunnel/configuration.go

```diff
@@ -463,6 +476,53 @@ func parseConfigIPVersion(version string) (v allregions.ConfigIPVersion, err err
 	return
 }
 
+func parseConfigBindAddress(ipstr string) (net.IP, error) {
+	// Unspecified - it's fine
+	if ipstr == "" {
+		return nil, nil
+	}
+	ip := net.ParseIP(ipstr)
+	if ip == nil {
+		return nil, fmt.Errorf("invalid value for edge-bind-address: %s", ipstr)
+	}
+	return ip, nil
+}
+
+func testIPBindable(ip net.IP) error {
+	var network, address string
+	if ip.To4() != nil {
+		network, address = "udp4", "127.0.0.1:4"
+	} else {
+		network, address = "udp6", "[::1]:4"
+	}
+
+	dialer := net.Dialer{LocalAddr: &net.UDPAddr{IP: ip}}
```

Comment:
Instead of a dialer, since cloudflared will be listening to this address, you can perform this test by listening to this address via `net.ListenUDP`. It also helps reduce the need to specify a port to attempt to dial to since we can let the OS determine which port to listen to.
Something along the lines of this: 
```go
addr, err := net.ResolveUDPAddr("udp", "[::1]:0") // provided edge-bind-address
listener, err := net.ListenUDP("udp", addr)
listener.Close()
```